src/Pirouette/Symbolic/Eval.hs

```diff
@@ -216,25 +219,34 @@ instance (SymEvalConstr lang) => Alternative (SymEval lang) where
               (runSymEvalRaw env st xs, runSymEvalRaw env st ys)
        in xs' <|> ys'
 
--- | Learn a new constraint and add it as a conjunct to the set of constraints of
---  the current path, then checks whether it leads to an inconsistency and fails
--- if it does.
-learn :: (SymEvalConstr lang) => [C.Constraint lang SymVar] -> SymEval lang ()
-learn cs = do
+pruneOrGetPathProblem :: (SymEvalConstr lang) => [C.Constraint lang SymVar] -> SymEval lang (SolverProblem lang Bool)
```

Comment:
`pruneOrGetPathProblem` takes the current state and outputs a (weighted) list of each possible new state obtained when adding one of the constraints taken as input; if the constraint is inconsistent with the state the branch is killed off and doesn't appear in the resulting list.